core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -97,6 +97,11 @@ private[spark] abstract class YarnSchedulerBackend(
     new YarnDriverEndpoint(rpcEnv, properties)
   }
 
+  override def stop(): Unit = {
+    requestTotalExecutors(0, 0, Map.empty)
```

Comment:
I think it would be better to add some documents to describe why we need to call `requestTotalExecutors` in stop.